### DISCUSSION TITLE
How I get the association in following example [Panache - Reactive - Kotlin]

### DISCUSSION THREAD

**Discussion by oberstrike** [quarkusio/quarkus#24859] [Q&A]
Hello folks,

I dont't know if this is the right channel to ask, but I have a question about Panache-Reactive-Kotlin. Maybe my question is very basic, but I just started to practice a little bit with panache reactive and mutiny/uni. I created a little demonstration project with following relations:

```
  @get:ManyToOne(fetch = FetchType.LAZY, optional = false)
    override var folder: FolderEntity? = null
```

and 

```
     @get:OneToMany(
        mappedBy = "folder",
        cascade = [CascadeType.ALL],
        fetch = FetchType.LAZY,
        orphanRemoval = true
    )
    override var dataFiles: MutableList<DataFileEntity> = mutableListOf()
```

I try to list all FolderEntities, convert them to a DTO with the IDs of the refered DataFiles.
My first try is this:

```
       val data = page.list<FolderEntity>()
            .map { folderEntities ->
                folderEntities.map { folderEntity ->
                    Mutiny.fetch(folderEntity.dataFiles).map { dataFiles ->
                        val show = folderEntity.toShow()
                        show.copy(
                            dataFiles = dataFiles.map { it.id },
                            size = dataFiles.size
                        )
                    }
                }
            }
```

Now I got the problem that the variable "data" has the type of Uni<List<Uni<FolderShow>>>
but I want Uni<List<FolderShow>>.

Can someone help me? Perhaps a reference to a thread or similar is enough.

kind regards
oberstrike

[a comment by an automated account is not shown]

**Comment by loicmathieu** [comment]
Usually, in fonctionnal programming when you have a List of things and want a thing of list you need to call flatMap, for Mutiny this is the same.
So you need to use a flatMap somewhere
